cw-orch/tests/entry_point.rs

```diff
@@ -0,0 +1,164 @@
+use mock_contract::{ExecuteMsg, InstantiateMsg, MigrateMsg, MockContract, QueryMsg};
+
+use cosmwasm_std::Event;
+use cw_orch::daemon::Daemon;
+use cw_orch::prelude::ContractInstance;
+use cw_orch::prelude::CwOrcExecute;
+use cw_orch::prelude::CwOrcMigrate;
+use cw_orch::prelude::CwOrcQuery;
+use std::sync::Arc;
+use tokio::runtime::Runtime;
+
+use cw_orch::prelude::CwOrcUpload;
+mod common;
+use cosmwasm_std::Addr;
+use cw_orch::prelude::{CwOrcInstantiate, Mock};
+
+#[test]
+fn test_instantiate() {
+    let contract = MockContract::new(
+        "test:mock_contract",
+        Mock::new(&Addr::unchecked("Ghazshag")).unwrap(),
+    );
+    contract.upload().unwrap();
+
+    contract
+        .instantiate(&InstantiateMsg {}, None, None)
+        .unwrap();
+}
+
+#[test]
+fn test_execute() {
+    let contract = MockContract::new(
+        "test:mock_contract",
+        Mock::new(&Addr::unchecked("Ghazshag")).unwrap(),
+    );
+    contract.upload().unwrap();
+
+    contract
+        .instantiate(&InstantiateMsg {}, None, None)
+        .unwrap();
+
+    let response = contract
+        .execute(&ExecuteMsg::FirstMessage {}, None)
+        .unwrap();
+    response.has_event(
+        &Event::new("wasm")
+            .add_attribute("_contract_addr", "contract0")
+            .add_attribute("action", "first message passed"),
+    );
+
+    contract
+        .execute(&ExecuteMsg::SecondMessage { t: "".to_string() }, None)
+        .unwrap_err();
+}
+
+#[test]
+fn test_query() {
+    let contract = MockContract::new(
+        "test:mock_contract",
+        Mock::new(&Addr::unchecked("Ghazshag")).unwrap(),
+    );
+    contract.upload().unwrap();
+
+    contract
+        .instantiate(&InstantiateMsg {}, None, None)
+        .unwrap();
+
+    let response: String = contract.query(&QueryMsg::FirstQuery {}).unwrap();
+    assert_eq!(response, "first query passed");
+
+    contract
+        .query::<String>(&QueryMsg::SecondQuery {})
+        .unwrap_err();
+}
+
+#[test]
+fn test_migrate() {
+    let admin = Addr::unchecked("Ghazshag");
+    let contract = MockContract::new("test:mock_contract", Mock::new(&admin).unwrap());
+    contract.upload().unwrap();
+
+    contract
+        .instantiate(&InstantiateMsg {}, Some(&admin), None)
+        .unwrap();
+
+    contract
+        .migrate(
+            &MigrateMsg {
+                t: "error".to_string(),
+            },
+            contract.code_id().unwrap(),
+        )
+        .unwrap_err();
+    let response = contract
+        .migrate(
+            &MigrateMsg {
+                t: "success".to_string(),
+            },
+            contract.code_id().unwrap(),
+        )
+        .unwrap();
+    assert_eq!(response.events.len(), 1);
+}
+
+#[test]
+#[cfg(feature = "node-tests")]
+fn daemon_test() {
+    use cw_orch::prelude::networks;
+
+    let runtime = Arc::new(Runtime::new().unwrap());
```

Comment:
this doesn't need to be Arc

Reply:
yes. I used what was used in other tests here